### DISCUSSION TITLE
How to tree shaking properly?

### DISCUSSION THREAD

**Discussion by jemikanegara** [vercel/next.js#17194] [Help]
My team use yarn workspace monorepo
-- Web package (Nextjs)
-- Components package (es6)

I have read that webpack tree shaking need to use es2015 module syntax (https://webpack.js.org/guides/tree-shaking/) but I have problem when build nextjs that importing the components package (got error on compiled server folder because it use commonjs)

Should I convert the server into es6? if that so how to do it? I tried using next/babel but it still produce commonjs for server

**Comment by nandorojo** [comment]
I'm also having some issues with this.

One suggestion that helped me a bit: add `sideEffects: false` to the `package.json` file to all packages in a monorepo (assuming they don't have side effects outside of their imports.)

**Reply by R-Bower** [reply]
This resolved the issue for my monorepo!

**Comment by leerob** [comment]
For anyone landing from Google: https://nextjs.org/docs/architecture/nextjs-compiler#modularize-imports
